### PR TITLE
refactor init assessment

### PR DESCRIPTION
**Pull request by minotogna**
Preparatory pr for #2905 

In order to let users to navigate between assessments, it is needed to have both assessments client side.
This pr includes the following:

- [x] BE: init endpoint : include all assessments and settings (to be able to navigate to default assessments from root page)
- [x] FE: refactor assessment store slice to store and access both assessments 
- [x] FE/BE: fetch metacache from Cycle page 
- [ ] FE/BE: remove getAssessment action / endpoint 

Future thinking: 
- move countries and region groups from assessment store slice to areas and store them by assessment/cycle ?
- rename assessment store slice to app ?

### CODE REVIEW

File: src/client/store/assessment/hooks/index.ts [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
+import { Section, SubSection } from 'meta/assessment'
 
-import { RootState, useAppSelector } from 'client/store'
+import { useAppSelector } from 'client/store'
 import { useCountryIso } from 'client/hooks'
 
-export { useCycle } from './useCycle'
-
 // TODO: Move elsewhere <>
 const getLocale = (isoCode: string): string => {
```
Comment (sorja) [inline src/client/store/assessment/hooks/index.ts:13]: Should we move this function? Where should it be located?

File: src/client/store/assessment/selectors/index.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { createSelector } from '@reduxjs/toolkit'
+
+import { AssessmentName } from 'meta/assessment'
+
+import { RootState } from 'client/store/RootState'
+
+const getAssessments = createSelector(
+  [(state: RootState) => state.assessment.assessments],
+  (assessments) => assessments
+)
```
Comment (sorja) [inline src/client/store/assessment/selectors/index.ts:10]: could this be just

```
const getAssessments = createSelector(
  (state: RootState) => state.assessment.assessments,
)
```

File: src/client/store/data/hooks/useDataSourcesLinked.ts [hunk 3]
```diff
@@ -4,8 +4,12 @@ import { useAppSelector } from 'client/store'
...
 export const useDataSourcesLinked = (props: { sectionName: string }): Array<DataSourceLinked> | undefined => {
+  const { sectionName } = props
+
   const assessment = useAssessment()
   const cycle = useCycle()
+  const assessmentName = assessment.props.name
+  const cycleName = cycle.name
 
-  return useAppSelector((state) => state.data[assessment.props.name][cycle.name].linkedDataSources[props.sectionName])
+  return useAppSelector((state) => state.data[assessmentName]?.[cycleName]?.linkedDataSources?.[sectionName])
```
Comment (sorja) [inline src/client/store/data/hooks/useDataSourcesLinked.ts:14]: should we use selector

File: src/client/store/assessment/selectors/index.ts (hunk 2 above)
Comment (minotogna) [reply src/client/store/assessment/selectors/index.ts:10]: Nope .

File: src/client/store/data/hooks/useDataSourcesLinked.ts (hunk 3 above)
Comment (minotogna) [reply src/client/store/data/hooks/useDataSourcesLinked.ts:14]: Not the scope of this pr . Let's keep things separated

File: src/client/store/assessment/hooks/index.ts (hunk 1 above)
Comment (minotogna) [reply src/client/store/assessment/hooks/index.ts:13]: Not the scope of this pr

**Review by sorja — APPROVED** [review]

**Pull request merged**